components/automate-ui/src/app/pages/+compliance/+reporting/+reporting-node/reporting-node.component.ts

```diff
@@ -205,4 +210,44 @@ export class ReportingNodeComponent implements OnInit, OnDestroy {
         this.activeReport = Object.assign(report, data);
       });
   }
+  
+  toggleDownloadDropdown() {
+    this.downloadOptsVisible = !this.downloadOptsVisible;
+  }
+  hideDownloadStatus() {
+    this.downloadStatusVisible = false;
+    this.downloadInProgress = false;
+    this.downloadFailed = false;
+  }
+
+  showDownloadStatus() {
+    this.downloadStatusVisible = true;
+    this.downloadInProgress = true;
+    this.downloadFailed = false;
+  }
+
+  onDownloadNodeReport(format) {
+    this.downloadOptsVisible = false;
+    const id: string = this.route.snapshot.params['id'];
+    const reportQuery = this.reportQueryService.getReportQuery();
+    reportQuery.filters = reportQuery.filters.concat([{type: {name: 'node_id'}, value: {id}}]);
+    const filename = `${reportQuery.endDate.format('YYYY-M-D')}.${format}`;
+
+    const onComplete = () => this.downloadInProgress = false;
+    const onError = _e => this.downloadFailed = true;
+    const types = { 'json': 'application/json', 'csv': 'text/csv' };
+    const onNext = data => {
+      const type = types[format];
+      const blob = new Blob([data], { type });
+      saveAs(blob, filename);
+      this.hideDownloadStatus();
+    };
+
+    this.downloadList = [filename];
+    this.showDownloadStatus();
+    this.statsService.downloaNodeReport(format, reportQuery).pipe(
```

Comment:
fix here and at the definition of course.
```suggestion
    this.statsService.downloadNodeReport(format, reportQuery).pipe(
```